ctapipe/io/simteleventsource.py

```diff
@@ -471,13 +476,17 @@ def _fill_trigger_info(data, array_event):
             data.trigger.event_type = EventType.UNKNOWN
 
         data.trigger.tels_with_trigger = trigger["triggered_telescopes"]
+        if self.allowed_tels:
+            data.trigger.tels_with_trigger = np.intersect1d(
+                data.trigger.tels_with_trigger, np.array(list(self.allowed_tels))
+            )
         central_time = parse_simtel_time(trigger["gps_time"])
         data.trigger.time = central_time
 
         for tel_id, time in zip(
-            trigger["triggered_telescopes"], trigger["trigger_times"]
+            data.trigger.tels_with_trigger, trigger["trigger_times"]
```

Comment:
Is this correct? I think you are messing up trigger times and telescope ids here.